x-pack/plugins/apm/public/components/shared/Links/MachineLearningLinks/MLLink.test.tsx

```diff
@@ -19,6 +19,6 @@ test('MLLink produces the correct URL', async () => {
   );
 
   expect(href).toMatchInlineSnapshot(
-    `"/app/ml/jobs?mlManagement=(groupIds:!(apm),jobId:!(something))&_g=(refreshInterval:(pause:!t,value:0),time:(from:now-5h,to:now-2h))"`
+    `"/app/ml/jobs?_a=(queryText:'groups:(apm)')&_g=(refreshInterval:(pause:!t,value:0),time:(from:now-5h,to:now-2h))"`
```

Comment:
Did the `jobId` disappear?

Reply:
@sqren, yes, now we store the entire query text in the URL, and the default filtering is based on the job id

Reply:
@sqren yes, and now I've realized we had an incorrect behavior before! The original code of restoring state from the URL was taking into account only one of the params, in particular 
```javascript
    let defaultQueryText;	
    const { jobId, groupIds } = getSelectedIdFromUrl(window.location.href);	

    if (groupIds !== undefined) {	
      defaultQueryText = getGroupQueryText(groupIds);	
    } else if (jobId !== undefined) {	
      defaultQueryText = getJobQueryText(jobId);	
    }
```
that's why I decided to change the URL generator itself. I'm going to fix it now

Reply:
Fixed in https://github.com/elastic/kibana/pull/83149/commits/593cd8344b607022112235c914ef6efa67211bd6